datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDataSourceFactory.java

```diff
@@ -46,11 +52,34 @@ public DataSource createDataSource(DataSourceMetadata metadata) {
 
   private void validateGlueDataSourceConfiguration(Map<String, String> dataSourceMetadataConfig)
       throws URISyntaxException, UnknownHostException {
-    DatasourceValidationUtils.validateLengthAndRequiredFields(
-        dataSourceMetadataConfig,
-        Set.of(GLUE_AUTH_TYPE, GLUE_ROLE_ARN, FLINT_URI, FLINT_REGION, FLINT_AUTH));
+    AuthenticationType authenticationType =
+        AuthenticationType.get(dataSourceMetadataConfig.get(GLUE_INDEX_STORE_OPENSEARCH_AUTH));
+    if (AuthenticationType.BASICAUTH.equals(authenticationType)) {
+      DatasourceValidationUtils.validateLengthAndRequiredFields(
+          dataSourceMetadataConfig,
+          Set.of(
+              GLUE_AUTH_TYPE,
+              GLUE_ROLE_ARN,
+              GLUE_INDEX_STORE_OPENSEARCH_URI,
+              GLUE_INDEX_STORE_OPENSEARCH_REGION,
```

Comment:
Why do we need region for basic auth? Shouldn't region only be for sig v4?

Reply:
You are right. will address that.